modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/SegmentsExperiments.es.js

```diff
@@ -68,6 +68,8 @@ function SegmentsExperiments({
 		: selectedExperienceId;
 	const noExperimentIllustration = `${assetsPath}${NO_EXPERIMENT_ILLUSTRATION_FILE_NAME}`;
 	const winnerVariant = variants.find((variant) => variant.winner === true);
+	const goalTarget =
+		experiment?.goal?.target && experiment.goal.target.substring(1);
```

Comment:
Don't combine `?.` and `&&` — the whole point of `?.` is to not have to use the `&&`.

Try something like:

```js
const goalTarget = experiment?.goal?.target?.substring(1);
```

(At the same time, why is this all so null-ish? I'm wondering why we have to worry about things potentially not being there at every single level.)

Reply:
You can or cannot have an experiment created for your experience. If you have an experiment, there is a goal (bounce rate or click) but maybe there is no target yet (only for click goal). And then the target comes with # (#myId) so I remove it to display in the ClayInput box 🤷‍♀️